motr/client_init.c

```diff
@@ -1604,6 +1613,47 @@ int m0_client_init(struct m0_client **m0c_p,
 	/* Init the hash-table for RM contexts */
 	rm_ctx_htable_init(&m0c->m0c_rm_ctxs, M0_RM_HBUCKET_NR);
 
+	if (ENABLE_DTM0) {
+		struct m0_reqh_service *reqh_svc;
+
+		rc = m0_conf_process2service_get(m0_reqh2confc(&m0c->m0c_reqh),
+						 &m0c->m0c_reqh.rh_fid,
+						 M0_CST_DTM0, &cli_svc_fid);
+		M0_ASSERT(rc == 0);
+
+		if (m0_dtm0_in_ut()) {
+			/* When in UT, m0c_reqh.rh_fid is the same as the
+			 * server process fid. It causes the client to use
+			 * the server-side service fid. Hard-coding of the client
+			 * service fid resolves this problem.
+			 * TODO: When in UT, walk over the list of processes in
+			 * the conf cache and get the service with "volatile"
+			 * property from there. It will help to get rid of the
+			 * hard-coded service fid.
+			 */
+			cli_svc_fid  = M0_FID_INIT(0x7300000000000001, 0x1a);
+		}
+
+		rc = m0_dtm_client_service_start(&m0c->m0c_reqh, &cli_svc_fid,
+						 &reqh_svc);
+		M0_ASSERT(rc == 0);
+
+		m0c->m0c_dtms = m0_dtm0_service_find(&m0c->m0c_reqh);
+		M0_ASSERT(m0c->m0c_dtms != NULL);
+
+		/* When in non-UT, we should wait until all the servers
+		 * have established connections with this client.
+		 * TODO: remove this sleep() when a proper start/stop
+		 * procedure is implemented.
+		 */
+		if (!m0_dtm0_in_ut()) {
+#ifndef __KERNEL__
+			sleep(15);
+#else
+			msleep(15000);
+#endif
```

Comment:
Please use m0_nanosleep() from lib/time.h

Reply:
Will be completely removed during working on DTM0 start/stop.